### PR TITLE
Chore: add missing/clean events

### PR DESCRIPTION
**Pull request by haythem96**
# Task: add missing/clean events

**Merge after #744**

## Description

Fixes ENG-829 ENG-1027 ENG-1026 ENG-1022

- Removed `TransferToOrder` event, because I don't think internal functions should emit events, and also this event is the same as the ERC20 events where address is oSQTH/WETH and sender == auction address. **(Fix N-14 issue in audit report)**
- Removed `TransferFromOrder` event, because I don't think internal functions should emit events, and also this event is the same as the ERC20 events where address is oSQTH/WETH and receiver == auction address. **(Fix N-14 issue in audit report)**
- Added more values in the `Deposit` event 
- Added Farm, DepositEthIntoCrab and WithdrawShutdown events. **(Fix N-14 issue in audit report)** 
- **(Fix N-01 issue in audit report)** 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Please describe how to test to verify the changes. Provide instructions so we can reproduce.

## FE Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

## User Facing Checklist

- [ ] I fully understand the user problem this PR is solving
- [ ] I know who the target user is for this PR and have a deep understanding of that user
- [ ] I have tried this flow thinking from the pov of the target user for this PR
- [ ] I (or working w someone on team) have scheduled a user test for this PR (if it is a large change)

### CODE REVIEW

File: packages/bull-vault/src/LeverageBull.sol [hunk 1]
```diff
@@ -56,7 +56,9 @@ contract LeverageBull is Ownable {
     /// @dev auction contract address
     address public auction;
 
-    event RepayAndWithdrawFromLeverage(address from, uint256 usdcToRepay, uint256 wethToWithdraw);
+    event AuctionRepayAndWithdrawFromLeverage(
+        address from, uint256 usdcToRepay, uint256 wethToWithdraw
```
Comment (alpinechicken) [inline packages/bull-vault/src/LeverageBull.sol:60]: indexed? seems to be the fashion

**Review by alpinechicken — APPROVED** [review]
lgtm

File: packages/bull-vault/src/BullStrategy.sol [hunk 2]
```diff
@@ -51,16 +51,27 @@ contract BullStrategy is ERC20, LeverageBull {
...
+        uint256 crabToRedeem,
+        uint256 wPowerPerpToRedeem,
+        uint256 usdcToRepay,
+        uint256 wethToWithdraw
+    );
+    event Deposit(address indexed from, uint256 crabAmount, uint256 wethLent, uint256 usdcBorrowed);
     event SetCap(uint256 oldCap, uint256 newCap);
     event RedeemCrabAndWithdrawEth(
-        uint256 crabToRedeem, uint256 wPowerPerpRedeemed, uint256 wethBalanceReturned
+        uint256 indexed crabToRedeem, uint256 wPowerPerpRedeemed, uint256 wethBalanceReturned
```
Comment (aleone) [inline packages/bull-vault/src/BullStrategy.sol:65]: why indexed here?

**Review by aleone — APPROVED** [review]
1 comment on unnecessary indexed imo

**Pull request merged**